src/main/java/teammates/storage/entity/Entity.java

```diff
@@ -0,0 +1,20 @@
+package teammates.storage.entity;
+
+import java.lang.reflect.Field;
+
+import javax.jdo.annotations.PrimaryKey;
+
+final class Entity {
```

Comment:
Yup that was my initial design haha but I was trying my best to avoid any PMD violations. Guess I will suppress the abstract class without abstract methods violation.